synapse/lib/types.py

```diff
@@ -346,8 +347,27 @@ class Int(Type):
     )
 
     def postTypeInit(self):
-        self.minval = self.opts.get('min')
-        self.maxval = self.opts.get('max')
+        self._size = self.opts.get('size')
+        self._signed = self.opts.get('signed')
+        minval = self.opts.get('min')
+        maxval = self.opts.get('max')
+
+        minmin, maxmax = -2 ** ((self._size * 8) - 1), 2 ** ((self._size * 8) - 1) - 1
+        if minval is None:
+            minval = minmin
+        if maxval is None:
+            maxval = maxmax
+        if minval < minmin or maxval > maxmax or maxval < minval:
```

Comment:
lol.  <3

Reply:
If any of these things is true, you're gonna have a bad time (at runtime)